core/app/c/[communitySlug]/stages/components/StageList.tsx

```diff
@@ -54,96 +51,113 @@ export async function StageList(props: Props) {
+					basePath={`/c/${pageContext.params.communitySlug}/stages`}
 				/>
 			</Suspense>
 		</div>
 	);
 }
 
-async function StagePubs({
+export async function StagePubs({
 	stage,
-	token,
-	stageById,
 	pageContext,
 	members,
+	totalPubLimit,
+	basePath,
+	pagination,
 }: {
-	stage: StagewithConstraints;
-	token: string;
-	stageById: StagesById;
+	stage: CommunityStage;
 	pageContext: PageContext;
 	members?: MemberWithUser[];
+	totalPubLimit?: number;
+	pagination?: { page: number; pubsPerPage: number };
+	basePath: string;
 }) {
 	const [stagePubs, actionInstances] = await Promise.all([
 		getPubs(
 			{ stageId: stage.id },
 			{
 				onlyParents: false,
-				limit: 100,
+				// fetch one extra pub so we know whether or not to render a show more button
+				limit: pagination?.pubsPerPage || (totalPubLimit && totalPubLimit + 1),
+				offset: pagination && (pagination.page - 1) * pagination.pubsPerPage,
+				orderBy: "updatedAt",
 			}
 		),
 		getStageActions(stage.id).execute(),
 	]);
 
-	const sources = moveConstraintSourcesForStage(stage, stageById);
-	const destinations = stage.moveConstraints.map((stage) => stageById[stage.destinationId]);
+	const totalPages =
+		stage.pubsCount && pagination ? Math.ceil(stage.pubsCount / pagination.pubsPerPage) : 0;
 
 	return (
 		<div className="flex flex-col gap-8">
-			{stagePubs.map((pub, index, list) => {
+			{stagePubs.map((pub, index) => {
+				if (totalPubLimit && index > totalPubLimit - 1) {
+					return null;
+				}
 				// this way we don't pass unecessary data to the client
 				const { children, ...basePub } = pub;
 				return (
-					<Fragment key={pub.id}>
-						<PubRow
-							key={pub.id}
-							pub={pub}
-							token={token}
-							actions={
-								<StagePubActions
-									key={stage.id}
-									moveFrom={sources}
-									moveTo={destinations}
-									pub={basePub}
-									stage={stage}
-									actionInstances={actionInstances}
-									pageContext={pageContext}
-									members={members}
-								/>
-							}
-							searchParams={pageContext.searchParams}
-						/>
-						{index < list.length - 1 && <hr />}
-					</Fragment>
+					<PubRow
+						key={pub.id}
+						pub={pub}
+						actions={
+							<StagePubActions
+								key={stage.id}
+								pub={basePub}
+								stage={stage}
+								actionInstances={actionInstances}
+								pageContext={pageContext}
+								members={members}
+							/>
+						}
+						searchParams={pageContext.searchParams}
+					/>
 				);
 			})}
+			{pagination && (
+				<BasicPagination
+					basePath={basePath}
+					searchParams={pageContext.searchParams}
+					page={pagination.page}
+					totalPages={totalPages}
+				/>
+			)}
+			{!pagination && totalPubLimit && stagePubs.length > totalPubLimit && (
+				<Button
+					variant="ghost"
+					className="text-md inline-flex text-muted-foreground"
+					size="lg"
+					asChild
+				>
+					<Link href={`stages/${stage.id}`}>See all pubs in stage {stage.name}</Link>
```

Comment:
I don't get why this link works? it should be `/c/${communitySlug}/stages/${stage.id}`, or, as i've learned, `./${stage.id}`, but somehow next just... figures it out?

I don't get it, the docs also don't mention anything about this behavior. For my sanity, could you change it to be `/c/${communitySlug}/stages/${stage.id}`?

Reply:
Yeah the way I did it here is just straight up wrong, no idea why it works. I was experiencing some very strange behavior with these links which is probably how I ended up getting this one wrong. I'll go ahead and make all these links relative to the root like you suggested!